Desktop/qquick/datasetview.h

```diff
@@ -55,20 +54,23 @@ class DataSetView : public QQuickItem
 	Q_PROPERTY( double					headerHeight			READ headerHeight											NOTIFY headerHeightChanged			)
 	Q_PROPERTY( double					rowNumberWidth			READ rowNumberWidth			WRITE setRowNumberWidth			NOTIFY rowNumberWidthChanged		)
 	Q_PROPERTY( bool					cacheItems				READ cacheItems				WRITE setCacheItems				NOTIFY cacheItemsChanged			)
+	Q_PROPERTY( bool					expandDataSet			READ expandDataSet			WRITE setExpandDataSet			NOTIFY expandDataSetChanged			)
 	Q_PROPERTY( QQuickItem			*	tableViewItem			READ tableViewItem			WRITE setTableViewItem												)
 	Q_PROPERTY( QItemSelectionModel *	selection				READ selectionModel											NOTIFY selectionModelChanged		)
-	Q_PROPERTY(	QModelIndex				selectionStart			READ selectionStart			WRITE setSelectionStart			NOTIFY selectionStartChanged		)
-	Q_PROPERTY(	QModelIndex				selectionEnd			READ selectionEnd			WRITE setSelectionEnd			NOTIFY selectionEndChanged			)
+	Q_PROPERTY(	QPoint					selectionStart			READ selectionStart			WRITE setSelectionStart			NOTIFY selectionStartChanged		)
+	Q_PROPERTY(	QPoint					selectionEnd			READ selectionEnd			WRITE setSelectionEnd			NOTIFY selectionEndChanged			)
 	Q_PROPERTY(	bool					editing					READ editing				WRITE setEditing				NOTIFY editingChanged				)
 	
 public:
+	friend ExpandDataProxyModel;
+
 							DataSetView(QQuickItem *parent = nullptr);
 
 	static DataSetView *	lastInstancedDataSetView()					{ return _lastInstancedDataSetView; }
 
 	void					setModel(QAbstractItemModel * model);
 	
-	QAbstractItemModel	*	model()								const	{ return _model;					}
+	QAbstractItemModel	*	model()								const	{ return _model->sourceModel();		}
```

Comment:
????

Reply:
I think Id prefer having both models stored as private variables, `_model` for the actual real model and `_expandModel` for your extension (which works great btw ^^)

Reply:
It is what I did first (_model & _expandModel as private), but I found it not satisfying: sometimes _model was used, sometimes _expandModel, I found it tricky. So I thought that all the call should go through the expanded Model, that wrapped completely the dataSetTableModel. 
Only the model property uses the dataSetTableModel: this is because the user of the DataSetView must set the dataSetTableModel as property. Maybe we could change the name of this property: innerModel or so....